soc/arm/nxp_imx/rt5xx/soc.h

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright (c) 2022, NXP
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @file
+ * @brief Board configuration macros for the MXRT595 platform
+ *
+ * This header file is used to specify and describe board-level aspects for the
+ * 'MXRT595' platform.
+ */
+
+#ifndef _SOC__H_
+#define _SOC__H_
+
+#ifndef _ASMLANGUAGE
+#include <sys/util.h>
+#include <fsl_common.h>
```

Comment:
these are not used here: files that require these headers should just include them.

Reply:
This is a bigger change as there a number of files in Zephyr including this file. Maybe this can be addressed as a separate PR.